app/src/main/java/com/github/onlynotesswent/ui/common/NoteComposables.kt

```diff
@@ -168,6 +163,100 @@ fun NoteItem(
         }
       }
 }
+
+/**
+ * Displays a bottom sheet with options to move or delete a note. The bottom sheet is displayed when
+ * the user clicks on the more options icon in the note card.
+ *
+ * @param note The note data that will be displayed in this card.
+ * @param noteViewModel The ViewModel that provides the list of notes to display.
+ * @param folderViewModel the folderViewModel used here to move the note.
+ * @param navigationActions The navigation instance used to transition between different screens.
+ * @param onDismiss The callback to be invoked when the bottom sheet is dismissed.
+ */
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun NoteOptionsBottomSheet(
+    note: Note,
+    noteViewModel: NoteViewModel,
+    folderViewModel: FolderViewModel,
+    navigationActions: NavigationActions,
+    onDismiss: () -> Unit
+) {
+  var showFileSystemPopup by remember { mutableStateOf(false) } // State to show FileSystemPopup
+  var showDeletePopup by remember { mutableStateOf(false) }
+
+  if (showFileSystemPopup) {
+    FileSystemPopup(
+        onDismiss = { showFileSystemPopup = false },
+        folderViewModel = folderViewModel,
+        onMoveHere = { selectedFolder ->
+          noteViewModel.updateNote(note.copy(folderId = selectedFolder?.id))
+          showFileSystemPopup = false
+          // this is needed to recompose the screen
```

Comment:
Why not just call navigate to folder contents with selected folder as the folderId parameter ? So you would move the note to selected folder and then dissmiss the pop up and navigate to the folder, which would in fact recompose the screen

Reply:
I did it this way as it is far less resource intensive to just recompose what is needed, here the grid with the notes than to recompose the whole screen. Looking back I think my comment was misleading and changed it.